src/applications/vaos/actions/appointments.js

```diff
@@ -289,17 +288,18 @@ export function cancelAppointment(appointment) {
   };
 }
 
-const SUBMITTED_REQUEST = 'Submitted';
 const CANCELLED_REQUEST = 'Cancelled';
 export function confirmCancelAppointment() {
   return async (dispatch, getState) => {
     const appointment = getState().appointments.appointmentToCancel;
-    const isPendingRequest = appointment.status === SUBMITTED_REQUEST;
+    const isConfirmedAppointment =
+      appointment.appointmentType === APPOINTMENT_TYPES.vaAppointment;
     const eventPrefix = `${GA_PREFIX}-cancel-appointment-submission`;
     const additionalEventdata = {
-      appointmentType: isPendingRequest ? 'pending' : 'confirmed',
-      facilityType: isCommunityCare(appointment) ? 'cc' : 'va',
+      appointmentType: !isConfirmedAppointment ? 'pending' : 'confirmed',
+      facilityType: appointment.isCommunityCare ? 'cc' : 'va',
     };
+    let apiData = appointment.apiData;
```

Comment:
Because we're transforming data, the regular appointment object is no longer identical to what vets-api sends us. We need the original data to cancel requests.

Ordinarily, the sort of transformation we're doing would make more sense in the selector layer, so that we don't have to keep a copy of the data around, but given that we're going to be transforming data for the FHIR work, we might as well use this approach now.